workerapi/worker.thrift

```diff
@@ -0,0 +1,38 @@
+enum Status {
+  UNKNOWN = 0
+  READY = 1
+  PENDING = 2
+  FAILURE = 3
+  RUNNING = 4
+  SUCCESS = 5
+  ABORTED = 6
+  TIMEOUT = 7
+}
+
+struct WorkerStatus {
+  1: required Status status
+  2: optional string taskId
+  3: optional string outUri
+  4: optional string errUri
+  5: optional string info
+  6: optional i32 exitCode
+}
+
+struct Task {
+  1: required list<string> argv
+  2: optional list<string> env
+  3: optional string dir
+  4: optional string taskId
```

Comment:
This would mean we can't batch schedule tasks because we won't know which taskID belongs to which task. Do we want batching?

Also, the scheduler is free to generate a new id if rerunning a failed task, although I personally prefer reusing the id since that gets us a single id for the entire task lifetime, and I think that's desirable for debugging at the very least. I'm not entirely following your logic and we should discuss this further offline.